src/pages/HomeAdministrator.vue

```diff
@@ -163,21 +159,76 @@ onMounted(() => {
   if (roarfirekit.value.restConfig) init();
 });
 
-const isSuperAdmin = computed(() => Boolean(userClaims.value?.claims?.super_admin));
+const orderBy = ref(orderByDefault);
+const fetchTestAdministrations = ref(false);
+const testAdminsCached = ref(false);
+
 const adminOrgs = computed(() => userClaims.value?.claims?.minimalAdminOrgs);
 const exhaustiveAdminOrgs = computed(() => userClaims.value?.claims?.adminOrgs);
-const orderBy = ref(orderByDefault);
+const isSuperAdmin = computed(() => Boolean(userClaims.value?.claims?.super_admin));
 const canQueryAdministrations = computed(() => {
   return initialized.value && !isLoadingClaims.value;
 });
+const showTestAdminsOnLabel = computed(() => {
+  return testAdminsCached.value ? 'Showing' : 'Fetching...';
+});
+
+/**
+ * Fetches administrations from the cache or the server
+ * @param {Array} queryKey - The query key to use for fetching the data
+ * @returns {Promise<unknown>} - The cached or fetched data
+ */
+const getAdministrations = async (queryKey) => {
+  let cachedData = await queryClient.getQueryData(queryKey);
+
+  if (!cachedData) {
+    cachedData = await queryClient.fetchQuery({
+      queryKey,
+      queryFn: () =>
+        administrationPageFetcher(
+          orderBy,
+          ref(10000),
+          ref(0),
+          isSuperAdmin,
+          adminOrgs,
+          exhaustiveAdminOrgs,
+          fetchTestAdministrations.value,
+        ),
+      keepPreviousData: true,
+      enabled: canQueryAdministrations,
+      staleTime: 5 * 60 * 1000, // 5 minutes
+    });
+    testAdminsCached.value = true;
+  }
+
+  return cachedData;
+};
+
+const { isLoading: isLoadingClaims, data: userClaims } = useQuery({
+  queryKey: ['userClaims', uid, userClaimsQueryKeyIndex],
+  queryFn: () => fetchDocById('userClaims', uid.value),
+  keepPreviousData: true,
+  enabled: initialized,
+  staleTime: 5 * 60 * 1000, // 5 minutes
+  cacheTime: Infinity,
+});
 
 const {
   isLoading: isLoadingAdministrations,
   isFetching: isFetchingAdministrations,
   data: administrations,
 } = useQuery({
   queryKey: ['administrations', uid, orderBy, ref(0), ref(10000), isSuperAdmin, administrationQueryKeyIndex],
```

Comment:
Should we add `fetchTestAdministrations` to the `queryKey`?

Reply:
Yes, I believe that we should add fetchTestAdministrations to the queryKey, since the query function depends on `fetchTestAdministrations`.

I did have to create new Firestore indexes for this based on `testData`, current org type, and name for `gse-roar-admin` and `gse-roar-admin-dev.`

Reply:
After testing with `fetchTestAdministrations` added to the queryKeys, I don't think that we want it in the queryKeys after all.

I can't fully explain the behavior, but including fetchTestAdministrations in the queryKey causes the test admins to display only briefly, then switch back to the non-test admins when toggling with the button. I believe that it has something to do with the watch function, but I am not sure how to describe it.

`watch(fetchTestAdministrations, async (newState) => {
  const queryKey = newState
    ? ['testAdministrations', uid, orderBy, ref(0), ref(10000), isSuperAdmin, administrationQueryKeyIndex]
    : ['administrations', uid, orderBy, ref(0), ref(10000), isSuperAdmin, administrationQueryKeyIndex];

  filteredAdministrations.value = await getAdministrations(queryKey);
});`